src/Avalonia.Controls.DataGrid/DataGrid.cs

```diff
@@ -2807,7 +2807,6 @@ internal void ProcessSelectionAndCurrency(int columnIndex, object item, int back
                         if (SelectionMode == DataGridSelectionMode.Extended && AnchorSlot != -1)
                         {
                             int anchorSlot = AnchorSlot;
-                            ClearRowSelection(slot, setAnchorSlot: false);
```

Comment:
Just a question though. Shouldn't `setAnchorSlot: false` actually disable resetting selection started index? I.e. exactly your issue. There could be an issue in the ClearRowSelection method instead.